### ISSUE TITLE
--list-all-pkgs does not show all packages

### ISSUE THREAD

**Issue by bgoareguer** [aquasecurity/trivy#1765]
## Description

I have a Python project with a `requirements.txt` file with the following content:
```
oauth2-client==4.0.0
waitress==1.2.1
python-gitlab==2.0.0
```

I run Trivy with the following command:
```
trivy --debug fs --list-all-pkgs --format json -o output.json ./requirements.txt
```

## What did you expect to happen?
I expected to see the 3 dependencies of my project in the `output.json` file, just as Syft does:
```
syft packages ./requirements.txt
 ✔ Indexed ./requirements.txt
 ✔ Cataloged packages      [3 packages]

NAME           VERSION  TYPE
oauth2-client  4.0.0    python
python-gitlab  2.0.0    python
waitress       1.2.1    python

```

## What happened instead?
Instead, Trivy only found the waitress package but not the 2 others:
```
trivy --debug fs --list-all-pkgs --format json -o output.json ./requirements.txt && cat output.json | jq .Results[0].Packages
2022-02-24T10:52:05.501Z        DEBUG   Severities: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
2022-02-24T10:52:05.513Z        DEBUG   cache dir:  /home/ubuntu/.cache/trivy
2022-02-24T10:52:05.516Z        DEBUG   DB update was skipped because the local DB is the latest
2022-02-24T10:52:05.517Z        DEBUG   DB Schema: 2, UpdatedAt: 2022-02-24 06:06:57.940258162 +0000 UTC, NextUpdate: 2022-02-24 12:06:57.940257962 +0000 UTC, DownloadedAt: 2022-02-24 09:38:59.660340316 +0000 UTC
2022-02-24T10:52:05.517Z        DEBUG   Vulnerability type:  [os library]
2022-02-24T10:52:05.527Z        DEBUG   OS is not detected and vulnerabilities in OS packages are not detected.
2022-02-24T10:52:05.528Z        DEBUG   Detected OS: unknown
2022-02-24T10:52:05.530Z        INFO    Number of language-specific files: 1
2022-02-24T10:52:05.530Z        INFO    Detecting pip vulnerabilities...
2022-02-24T10:52:05.530Z        DEBUG   Detecting library vulnerabilities, type: pip, path: requirements.txt
[
  {
    "Name": "waitress",
    "Version": "1.2.1",
    "Layer": {}
  }
]

```

## Output of `trivy -v`:

```
Version: 0.24.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-02-24 06:06:57.940258162 +0000 UTC
  NextUpdate: 2022-02-24 12:06:57.940257962 +0000 UTC
  DownloadedAt: 2022-02-24 09:38:59.660340316 +0000 UTC

```

## Additional details (base image name, container registry info...):
I might be wrong but it seems that Trivy only lists packages that contain vulnerabilities.

**Comment by afdesk** [comment]
@bgoareguer thanks for your report! I could reproduce this issue. 
and it seems that there is a problem if a package contains hyphen...
I'll try to investigate it

**Comment by afdesk** [comment]
@bgoareguer thanks! 
we fixed this issue, and a new version `v0.24.1` was released.
